src/cc_catalog_airflow/dags/provider_api_scripts/common/licenses/constants.py

```diff
@@ -0,0 +1,138 @@
+NO_VERSION = 'N/A'
+
+_SIMPLE_LICENSE_PATHS = [
+    # This list holds recognized license URL path snippets that split
+    # correctly into license_ and license_version, and for which we can
+    # recover the path from a valid license_, license_version pair.
+    'licenses/by/1.0',
+    'licenses/by/2.0',
+    'licenses/by/2.5',
+    'licenses/by/3.0',
+    'licenses/by/4.0',
+
+    'licenses/by-nc/1.0',
+    'licenses/by-nc/2.0',
+    'licenses/by-nc/2.5',
+    'licenses/by-nc/3.0',
+    'licenses/by-nc/4.0',
+
+    'licenses/by-nc-nd/2.0',
+    'licenses/by-nc-nd/2.5',
+    'licenses/by-nc-nd/3.0',
+    'licenses/by-nc-nd/4.0',
+
+    'licenses/by-nc-sa/1.0',
+    'licenses/by-nc-sa/2.0',
+    'licenses/by-nc-sa/2.5',
+    'licenses/by-nc-sa/3.0',
+    'licenses/by-nc-sa/4.0',
+
+    'licenses/by-nd/1.0',
+    'licenses/by-nd/2.0',
+    'licenses/by-nd/2.5',
+    'licenses/by-nd/3.0',
+    'licenses/by-nd/4.0',
+
+    'licenses/by-sa/1.0',
+    'licenses/by-sa/2.0',
+    'licenses/by-sa/2.5',
+    'licenses/by-sa/3.0',
+    'licenses/by-sa/4.0',
+
+    'licenses/devnations/2.0',
+
+    'licenses/nc/1.0',
+    'licenses/nc/2.0/jp',
+
+    'licenses/nc-sa/1.0',
+    'licenses/nc-sa/2.0/jp',
+
+    'licenses/nc-sampling+/1.0',
+
+    'licenses/nd/1.0',
+    'licenses/nd/2.0/jp',
+
+    'licenses/nd-nc/1.0',
+    'licenses/nd-nc/2.0/jp',
+
+    'licenses/sa/1.0',
+    'licenses/sa/2.0/jp',
+
+    'licenses/sampling+/1.0',
+    'licenses/sampling/1.0',
+]
+
+_SIMPLE_IRREVERSIBLE_LICENSE_PATHS = [
```

Comment:
These are irreversible because they cannot be recovered from just a (license, version) pair.  We would need the country info in order to have enough to determine which path is the correct one.

For the other direction, we can split them into the license, version pair which is used in CC Search in the following way:
![image](https://user-images.githubusercontent.com/6829750/87566405-46fb0600-c6c3-11ea-8f2b-a426117d14df.png)

Note that in the case that we're handed only a (license_, version) pair for one of these paths that have no unported version, we'll throw out the entire entry, since we can't recover a valid license URL in that case.